File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[]}
 */
var postorder = function(root) {
    const res = []
    dfs(root)
    return res
    
    function dfs(root) {
        if (!root) return
        for (const child of root.children) {
            dfs(child)
        }
        res.push(root.val)
    }
};

postorder = function(root, res = []) {
    if (!root) return res
    for (const child of root.children) {
        postorder(child, res)
    }
    res.push(root.val)
    return res
}

File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        if not root:
            return []
        stack, res = [root], []
        while stack:
            root = stack[-1]
            if root.children:
                while root.children:
                    stack.append(root.children.pop())
            else:
                res.append(root.val)
                stack.pop()
        return res
        
        # if not root:
        #     return []
        # res = []
        # for child in root.children:
        #     for sub in self.postorder(child):
        #         res.append(sub)
        # res.append(root.val)
        # return res